boot/bootutil/src/loader.c

```diff
@@ -1818,16 +1985,16 @@ split_go(int loader_slot, int split_slot, void **entry)
     if (sectors == NULL) {
         return SPLIT_GO_ERR;
     }
-    boot_data.imgs[loader_slot].sectors = sectors + 0;
-    boot_data.imgs[split_slot].sectors = sectors + BOOT_MAX_IMG_SECTORS;
+    boot_data->imgs[loader_slot].sectors = sectors + 0;
```

Comment:
Perhaps the `imgs` field in the `boot_loader_state` struct should be changed to `slots` or something? Currently it is quite confusing to be operating with two levels of `image`/`img`. Especially since it actually is slots not images being indexed there.

Reply:
I think it really makes sense, however I would appreciate it if it would be part of another following pull request. I'm afraid it would generate even more conflicts in this PR.

Reply:
Makes sense.